include/server.h

```diff
@@ -49,6 +50,11 @@ namespace kiwix
         * Stop the daemon.
         */
        void stop();
+       
+       /**
+        *  Get the best public IP from available interfaces given by getNetworkInterfaces()
+        */
+       std::string getBestPublicIp();
```

Comment:
This should not be the "best public IP".
This should be the best public IP where we can connect to the server.

We need a `getBestPublicIp` which get the best IP from networkInterface (At least internally. But it could be public to help UI decide what is the best default value) but it is not what we need here.
If the server is started with a specific address, we want to have this specific address back (else it it useless).
If the server is started without address, we indeed want to the "best" address.